docs/blog/2021-07-30-0.4.0-release.md

```diff
@@ -0,0 +1,87 @@
+
+*This is the initial implementation of the hook.  There are various limitations and future plans for transactions, please [read the docs](/docs/api-reference/core/useRecoilTransaction#current-limitations-and-future-vision).*
+
+**Example**
+
+Suppose we have two atoms, `positionState` and `headingState`, and we'd like to update them together as part of a single action, where the new value of `positionState` is a function of *both* the current value of `positionState` and `headingState`.  You can accomplish this with a transaction, which must be a pure function without side-effects:
+
+```jsx
+const goForward = useRecoilTransaction_UNSTABLE(({get, set}) => (distance) => {
+  const heading = get(headingState);
+  const position = get(positionState);
+  set(positionAtom, {
+    x: position.x + cos(heading) * distance,
+    y: position.y + sin(heading) * distance,
+  });
+});
+```
+
+Then you can execute the transaction by just calling `goForward(distance)` in an event handler.  This will update state based on the *current* values, not the state when the components rendered.  You can also read the values of previous writes during a transaction.  Because no other updates will be committed while the updater is executing, you will see a consistent store of state.
+
+the previous approach using [`useRecoilCallback()`](/docs/api-reference/core/useRecoilCallback) might have looked like the following:
+
+```jsx
+const goForward = useRecoilCallback(({snapshot, gotoSnapshot}) => (distance) => {
+  const mutatedSnapshot = snapshot.map(({get, set}) => {
+    const heading = get(headingState);
+    const position = get(positonState);
+    set(positionState, {
+      x: position.x + cos(heading) * distance,
+      y: position.y + sin(heading) * distance,
+    });
+  });
+  gotoSnapshot(mutatedSnapshot);
+});
+```
+
+This has the following drawbacks:
+* There is performance overhead for managing the full generality of snapshots.
+* There is more opportunity for bugs:  The snapshot might be retained and used in the future.  Since a snapshot contains the complete set of Recoil state, not just a changeset, that could accidentally rewind changes that occurred between creating and committing the snapshot.
+
+**Reducer Example**
+
+You can also use this hook to create a reducer pattern of executing actions over multiple atoms:
+
+```jsx
+const reducer = useRecoilTransaction_UNSTABLE(({get, set}) => action => {
+  switch(action.type) {
+    case 'goForward':
+      const heading = get(headingState);
+      set(positionState, position => {
+        x: position.x + cos(heading) * action.distance,
+        y: position.y + sin(heading) * action.distance,
+      });
+      break;
+
+    case 'turn':
+      set(headingState, action.heading);
+      break;
+  }
+});
+```
+
+## Selector Cache Policy Configuration
```

Comment:
I'd put this first, it's easier to explain (thus shorter) and probably of interest to more people.

Reply:
Yeah, good idea, was just thinking that also!